tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,321 @@
+                            onClick={() => act("material_eject", { "resource": resourceData.id })}
+                          />,
+                          <Button
+                            key="swap"
+                            icon="arrows-up-down"
+                            color={(swap !== resourceData.id) ? null : "green"}
+                            onClick={() => swapPriority(resourceData.id)}
+                          />,
+                        ]}
+                        label={toTitleCase(resourceData.name)}
+                        textAlign="center"
+                      >
+                        {resourceData.amount}
+                      </LabeledList.Item>
+                    ))}
+                  </LabeledList>
+                </Section>
+              </Stack.Item>
+              <Stack.Item>
+                <Section
+                  textAlign="center"
+                  title={"Fabricator Settings"}
+                >
+                  <LabeledList>
+                    <LabeledList.Item
+                      label={"Repeat"}
+                      buttons={<Button icon="repeat" onClick={() => toggleRepeat()}>Toggle Repeat</Button>}
+                      textAlign="center"
+                    >
+                      {data.repeat ? "On" : "Off"}
+                    </LabeledList.Item>
+                    <LabeledList.Item
+                      label="Speed"
+                    >
+                      <Slider
+                        minValue={1}
+                        value={data.speed}
+                        maxValue={data.hacked ? data.max_speed_hacked : data.max_speed_normal}
+                        step={1}
+                        stepPixelSize={100}
+                        disabled={data.mode !== "working"}
+                        onChange={(_e: any, value: number) => act("speed", { "value": value })}
+                      />
+                    </LabeledList.Item>
+                  </LabeledList>
+                </Section>
+                {data.manudrive.limit !== null ? (
+                  <Stack.Item>
+                    <Section
+                      title="Loaded Manudrive"
+                      buttons={
+                        <Button
+                          icon="eject"
+                          content="Eject"
+                          disabled={data.mode !== "ready"}
+                          onClick={() => act("manudrive", { "action": "eject" })}
+                        />
+                      }
+                      mb={1}
+                    >
+                      {data.manudrive.name}
+                      <Divider />
+                      <LabeledList>
+                        <LabeledList.Item
+                          label={"Fabrication Limit"}
+                        >
+                          {data.manudrive.limit === -1 ? "Unlimited" : `${data.manudrive.limit} ${pluralize("use", data.manudrive.limit)}`}
```

Comment:
General note that using `-1` to mean "unlimited" is somewhat risky, as it conceivably could decrement to `-1` without much going wrong code logic wise. Prefer `null` for no limit, or even a different property.

Reply:
this is how manufacturer drives do it, i would prefer to not expand scope by rewriting that functionality if possible, though i could put a ternary on the ui data

Reply:
also just to note there is a lot of safety/sanity checks on the dm side for this so it should be fine..? its the value defined for it but ill add it as a constant as well